# Introduction to Data Visualization - Assignment 1

SAMARASIMHA REDDY CHITTAMURU \
202289731 \
20 January 2023


## Data Preparation :

Before solving the questions let's see what we can do to prepare our data
1. Import the pandas Library in the Jupyter Notebook
2. Load the data into a Dataframe (from csv) 

In [1]:
## importing pandas library
import pandas as pd 

In [2]:
## Loading the data set
data_sw = pd.read_csv("starwars.csv")

## Data Cleaning : 

Now let's see what kind of data we have and the attribute and the missing values 

In [4]:
data_sw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1186 entries, 0 to 1185
Data columns (total 38 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               1186 non-null   float64
 1   seen_any_films                   1186 non-null   object 
 2   fan                              836 non-null    object 
 3   seen_episode1                    936 non-null    object 
 4   seen_episode2                    936 non-null    object 
 5   seen_episode3                    936 non-null    object 
 6   seen_episode4                    936 non-null    object 
 7   seen_episode5                    936 non-null    object 
 8   seen_episode6                    936 non-null    object 
 9   rank_episode1                    835 non-null    float64
 10  rank_episode2                    836 non-null    float64
 11  rank_episode3                    835 non-null    float64
 12  rank_episode4       

We can see that there are many columns with null values. This is because it's a survey and there can be many questions which are not applicable to a certain person taking the survey.


I choose to use the method posted in the bright space i.e... keeping the na values and carry out the analysis.

In [91]:
data_starwars = pd.read_csv("starwars.csv",keep_default_na = False)

In [8]:
data_starwars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1186 entries, 0 to 1185
Data columns (total 38 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   id                               1186 non-null   float64
 1   seen_any_films                   1186 non-null   object 
 2   fan                              1186 non-null   object 
 3   seen_episode1                    1186 non-null   object 
 4   seen_episode2                    1186 non-null   object 
 5   seen_episode3                    1186 non-null   object 
 6   seen_episode4                    1186 non-null   object 
 7   seen_episode5                    1186 non-null   object 
 8   seen_episode6                    1186 non-null   object 
 9   rank_episode1                    1186 non-null   object 
 10  rank_episode2                    1186 non-null   object 
 11  rank_episode3                    1186 non-null   object 
 12  rank_episode4       

Now we can see there are no  null values in the data set

### Question 1

In [14]:
data_starwars.age.nunique()

5

We can see it has 5 unique values. Let's dig deeper to see what they are.

In [23]:
data_starwars.age.unique()

array(['18-29', 'N/A', '30-44', '> 60', '45-60'], dtype=object)

It also included the N/A as a value. But that shouldn't be counted. 

##### Unique values of age :
Thus technically we have 4 age groups but there is also a value 'N/A'.

### Question 2

In [28]:
data_starwars.who_shot_first.unique().size

4

In [105]:
data_starwars.who_shot_first.unique()

array(["I don't understand this question", 'N/A', 'Greedo', 'Han'],
      dtype=object)

##### Unique response to the question 'who shot first' :
Thus from what we can see there are 3 responses and an N/A which can be opted out as an answer

### Question 3

In [38]:
darth = data_starwars[(data_starwars["view_darth_vader"]=="Very favorably") | (data_starwars["view_darth_vader"]=="Somewhat favorably")]

In [39]:
darth

id seen_any_films  fan seen_episode1 seen_episode2  \
0     3.292880e+09            Yes  Yes           Yes           Yes   
3     3.292763e+09            Yes  Yes           Yes           Yes   
4     3.292731e+09            Yes  Yes           Yes           Yes   
5     3.292719e+09            Yes  Yes           Yes           Yes   
6     3.292685e+09            Yes  Yes           Yes           Yes   
...            ...            ...  ...           ...           ...   
1175  3.288401e+09            Yes  Yes           Yes           Yes   
1180  3.288390e+09            Yes  Yes           Yes           Yes   
1181  3.288389e+09            Yes  Yes           Yes           Yes   
1184  3.288373e+09            Yes  Yes           Yes           Yes   
1185  3.288373e+09            Yes   No           Yes           Yes   

     seen_episode3 seen_episode4 seen_episode5 seen_episode6 rank_episode1  \
0              Yes           Yes           Yes           Yes             3   
3              Yes           Yes           Yes           Yes             5   
4              Yes           Yes           Yes           Yes             5   
5              Yes           Yes           Yes           Yes             1   
6              Yes           Yes           Yes           Yes             6   
...            ...           ...           ...           ...           ...   
1175           Yes           Yes           Yes           Yes             1   
1180           Yes           Yes           Yes           Yes             3   
1181           Yes           Yes           Yes           Yes             5   
1184           Yes           Yes           Yes           Yes             4   
1185            No            No           Yes           Yes             6   

      ...           view_yoda                    who_shot_first  \
0     ...      Very favorably  I don't understand this question   
3     ...      Very favorably  I don't understand this question   
4     ...  Somewhat favorably                            Greedo   
5     ...      Very favorably                               Han   
6     ...      Very favorably                               Han   
...   ...                 ...                               ...   
1175  ...  Somewhat favorably                               Han   
1180  ...      Very favorably                               Han   
1181  ...      Very favorably                               Han   
1184  ...      Very favorably                               Han   
1185  ...    Very unfavorably  I don't understand this question   

     familiar_with_expanded_universe fan_expanded_universe fan_star_trek  \
0                                Yes                    No            No   
3                                 No                   N/A           Yes   
4                                Yes                    No            No   
5                                Yes                    No           Yes   
6                                Yes                    No            No   
...                              ...                   ...           ...   
1175                             Yes                    No           Yes   
1180                              No                   N/A            No   
1181                              No                   N/A           Yes   
1184                              No                   N/A           Yes   
1185                              No                   N/A            No   

      gender    age     household_income                         education  \
0       Male  18-29                  N/A                High school degree   
3       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
4       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
5       Male  18-29    $25,000 - $49,999                   Bachelor degree   
6       Male  18-29                  N/A                High school degree   
...      ...    ...                  ...            

In [41]:
len(darth)

481

##### Number of entries in the data set that view Darth Vader as "Very favorably" or "Somewhat favorably :

Thus we have 481 entries which has answeres Very favorablt or Somewhat favorably to Darth Vader

### Question 4

In [78]:
all_seen = data_starwars[(data_starwars["seen_episode1"] == "Yes") & (data_starwars["seen_episode2"] == "Yes") & (data_starwars["seen_episode3"] == "Yes") & (data_starwars["seen_episode4"] == "Yes") & (data_starwars["seen_episode5"] == "Yes") & (data_starwars["seen_episode6"] == "Yes")]

In [106]:
all_seen

id seen_any_films  fan seen_episode1 seen_episode2  \
0     3.292880e+09            Yes  Yes           Yes           Yes   
3     3.292763e+09            Yes  Yes           Yes           Yes   
4     3.292731e+09            Yes  Yes           Yes           Yes   
5     3.292719e+09            Yes  Yes           Yes           Yes   
6     3.292685e+09            Yes  Yes           Yes           Yes   
...            ...            ...  ...           ...           ...   
1175  3.288401e+09            Yes  Yes           Yes           Yes   
1180  3.288390e+09            Yes  Yes           Yes           Yes   
1181  3.288389e+09            Yes  Yes           Yes           Yes   
1182  3.288379e+09            Yes  Yes           Yes           Yes   
1184  3.288373e+09            Yes  Yes           Yes           Yes   

     seen_episode3 seen_episode4 seen_episode5 seen_episode6 rank_episode1  \
0              Yes           Yes           Yes           Yes             3   
3              Yes           Yes           Yes           Yes             5   
4              Yes           Yes           Yes           Yes             5   
5              Yes           Yes           Yes           Yes             1   
6              Yes           Yes           Yes           Yes             6   
...            ...           ...           ...           ...           ...   
1175           Yes           Yes           Yes           Yes             1   
1180           Yes           Yes           Yes           Yes             3   
1181           Yes           Yes           Yes           Yes             5   
1182           Yes           Yes           Yes           Yes             4   
1184           Yes           Yes           Yes           Yes             4   

      ...           view_yoda                    who_shot_first  \
0     ...      Very favorably  I don't understand this question   
3     ...      Very favorably  I don't understand this question   
4     ...  Somewhat favorably                            Greedo   
5     ...      Very favorably                               Han   
6     ...      Very favorably                               Han   
...   ...                 ...                               ...   
1175  ...  Somewhat favorably                               Han   
1180  ...      Very favorably                               Han   
1181  ...      Very favorably                               Han   
1182  ...      Very favorably  I don't understand this question   
1184  ...      Very favorably                               Han   

     familiar_with_expanded_universe fan_expanded_universe fan_star_trek  \
0                                Yes                    No            No   
3                                 No                   N/A           Yes   
4                                Yes                    No            No   
5                                Yes                    No           Yes   
6                                Yes                    No            No   
...                              ...                   ...           ...   
1175                             Yes                    No           Yes   
1180                              No                   N/A            No   
1181                              No                   N/A           Yes   
1182                              No                   N/A           Yes   
1184                              No                   N/A           Yes   

      gender    age     household_income                         education  \
0       Male  18-29                  N/A                High school degree   
3       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
4       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
5       Male  18-29    $25,000 - $49,999                   Bachelor degree   
6       Male  18-29                  N/A                High school degree   
...      ...    ...                  ...            

In [107]:
len(all_seen)

471

##### Total number of people that have seen all 6 Star Wars films

We can see there are 471 entries who watched all 6 

In [76]:
episodes = ['seen_episode1','seen_episode2','seen_episode3','seen_episode4','seen_episode5','seen_episode6']

In [79]:
all_seen_1 = data_starwars[(data_starwars[episodes] == "Yes")]

In [84]:
all_seen_1[episodes].dropna()

seen_episode1 seen_episode2 seen_episode3 seen_episode4 seen_episode5  \
0              Yes           Yes           Yes           Yes           Yes   
3              Yes           Yes           Yes           Yes           Yes   
4              Yes           Yes           Yes           Yes           Yes   
5              Yes           Yes           Yes           Yes           Yes   
6              Yes           Yes           Yes           Yes           Yes   
...            ...           ...           ...           ...           ...   
1175           Yes           Yes           Yes           Yes           Yes   
1180           Yes           Yes           Yes           Yes           Yes   
1181           Yes           Yes           Yes           Yes           Yes   
1182           Yes           Yes           Yes           Yes           Yes   
1184           Yes           Yes           Yes           Yes           Yes   

     seen_episode6  
0              Yes  
3              Yes  
4              Yes  
5              Yes  
6              Yes  
...            ...  
1175           Yes  
1180           Yes  
1181           Yes  
1182           Yes  
1184           Yes  

[471 rows x 6 columns]

This is an alternative solution.

### Question_5

In [94]:
fans = data_starwars[data_starwars["fan"] == "Yes"]

In [95]:
fans

id seen_any_films  fan seen_episode1 seen_episode2  \
0     3.292880e+09            Yes  Yes           Yes           Yes   
3     3.292763e+09            Yes  Yes           Yes           Yes   
4     3.292731e+09            Yes  Yes           Yes           Yes   
5     3.292719e+09            Yes  Yes           Yes           Yes   
6     3.292685e+09            Yes  Yes           Yes           Yes   
...            ...            ...  ...           ...           ...   
1179  3.288392e+09            Yes  Yes            No            No   
1180  3.288390e+09            Yes  Yes           Yes           Yes   
1181  3.288389e+09            Yes  Yes           Yes           Yes   
1182  3.288379e+09            Yes  Yes           Yes           Yes   
1184  3.288373e+09            Yes  Yes           Yes           Yes   

     seen_episode3 seen_episode4 seen_episode5 seen_episode6 rank_episode1  \
0              Yes           Yes           Yes           Yes             3   
3              Yes           Yes           Yes           Yes             5   
4              Yes           Yes           Yes           Yes             5   
5              Yes           Yes           Yes           Yes             1   
6              Yes           Yes           Yes           Yes             6   
...            ...           ...           ...           ...           ...   
1179            No            No           Yes           Yes             3   
1180           Yes           Yes           Yes           Yes             3   
1181           Yes           Yes           Yes           Yes             5   
1182           Yes           Yes           Yes           Yes             4   
1184           Yes           Yes           Yes           Yes             4   

      ...           view_yoda                    who_shot_first  \
0     ...      Very favorably  I don't understand this question   
3     ...      Very favorably  I don't understand this question   
4     ...  Somewhat favorably                            Greedo   
5     ...      Very favorably                               Han   
6     ...      Very favorably                               Han   
...   ...                 ...                               ...   
1179  ...      Very favorably                               Han   
1180  ...      Very favorably                               Han   
1181  ...      Very favorably                               Han   
1182  ...      Very favorably  I don't understand this question   
1184  ...      Very favorably                               Han   

     familiar_with_expanded_universe fan_expanded_universe fan_star_trek  \
0                                Yes                    No            No   
3                                 No                   N/A           Yes   
4                                Yes                    No            No   
5                                Yes                    No           Yes   
6                                Yes                    No            No   
...                              ...                   ...           ...   
1179                              No                   N/A           Yes   
1180                              No                   N/A            No   
1181                              No                   N/A           Yes   
1182                              No                   N/A           Yes   
1184                              No                   N/A           Yes   

      gender    age     household_income                         education  \
0       Male  18-29                  N/A                High school degree   
3       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
4       Male  18-29  $100,000 - $149,999  Some college or Associate degree   
5       Male  18-29    $25,000 - $49,999                   Bachelor degree   
6       Male  18-29                  N/A                High school degree   
...      ...    ...                  ...            

In [103]:
fans["gender"].value_counts(normalize=True)

Male      0.548913
Female    0.431159
N/A       0.019928
Name: gender, dtype: float64

##### percentage of males and females that are fans of Star Wars :
The Male fans consists of 54.8913% while the Female's are 43.1159% and the remaining 0.019928% are N/A

In [104]:
data_starwars.groupby('fan')['gender'].value_counts(normalize=True)

fan  gender
N/A  Female    0.434286
     N/A       0.354286
     Male      0.211429
No   Female    0.559859
     Male      0.422535
     N/A       0.017606
Yes  Male      0.548913
     Female    0.431159
     N/A       0.019928
Name: gender, dtype: float64

This is an alternative solution

### Question 6

I have implemented the ask and used markdown to do the same.